src/Psalm/Internal/Codebase/AssertionsFromInheritanceResolver.php

```diff
@@ -61,80 +57,9 @@ public function resolve(
              * Since the inheritance does not provide its own assertions, we have to detect those
              * from inherited classes
              */
-            $assertions += array_map(
-                fn(Possibilities $possibilities) => $this->modifyAssertionsForInheritance(
-                    $possibilities,
-                    $this->codebase,
-                    $called_class,
-                    $inherited_classes_and_interfaces,
-                ),
-                $potential_assertion_providing_method_storage->assertions,
-            );
+            $assertions += $potential_assertion_providing_method_storage->assertions;
         }
 
         return $assertions;
     }
-
-    /**
```

Comment:
By removing https://github.com/vimeo/psalm/blob/f782767438dd1678e60f3f97cfc66e1252e79b10/src/Psalm/Internal/Analyzer/Statements/Expression/Call/ClassTemplateParamCollector.php#L51-L53, these LoC are not necessary anymore.